akka-http-proxy/src/main/scala/com/pagerduty/akka/http/proxy/HttpProxy.scala

```diff
@@ -39,6 +39,8 @@ class HttpProxy[AddressingConfig](
                         elapsed,
                         "upstream" -> upstream.metricsTag)
 
+      upstream.transformResponse(request, r)
```

Comment:
Akka models like `HttpResponse` are [immutable](https://en.wikipedia.org/wiki/Immutable_object), meaning they can't be changed after creation. This is true of many data structures in Scala, and it takes a little while to get used to.

What this means here is that if you want to transform (change) the response, it's actually more like a copy operation. And that means you need to save the (changed) copy somewhere to use it. Right now, this line just calls the `transformResponse` method and throws away the result (there's likely a warning about this from the compiler).

What we need to do instead is return this transformed response to the caller of this method. Right now the return value is below: `r.withEntity(e)` which means "take r (the response) and give it the entity we consumed". This is probably a good place to call the transformer, since it only needs to be invoked if we can actually consume the whole response:

``` scala
upstream.transformResponse(preparedProxyRequest, r.withEntity(e))
```

Note that you want to use `preparedProxyRequest`, not `request`, since the former is the request we're actually sending upstream.

Does that help?